commons/runtime/src/main/java/com/github/mcollovati/quarkus/hilla/security/RouteUtil.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2025 Marco Collovati, Dario Götze
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.github.mcollovati.quarkus.hilla.security;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Predicate;
+
+import com.vaadin.flow.internal.menu.MenuRegistry;
+import com.vaadin.flow.server.VaadinService;
+import com.vaadin.flow.server.menu.AvailableViewInfo;
+import com.vaadin.flow.server.startup.ApplicationConfiguration;
+import io.quarkus.security.identity.SecurityIdentity;
+import io.quarkus.vertx.http.runtime.security.ImmutablePathMatcher;
+import io.vertx.ext.web.RoutingContext;
+
+public class RouteUtil {
+
+    private Map<String, AvailableViewInfo> registeredRoutes = null;
+    private final VaadinService vaadinService;
+
+    public RouteUtil(VaadinService vaadinService) {
+        this.vaadinService = vaadinService;
+    }
+
+    public boolean isRouteAllowed(RoutingContext context, SecurityIdentity identity) {
+        if (registeredRoutes == null) {
+            collectClientRoutes();
+        }
+        var viewConfig = getRouteData(context.normalizedPath(), identity);
+        return viewConfig.isPresent();
+    }
+
+    private void collectClientRoutes() {
+        ApplicationConfiguration config = ApplicationConfiguration.get(vaadinService.getContext());
+        setRoutes(MenuRegistry.collectClientMenuItems(false, config, null));
```

Comment:
`MenuRegistry.collectClientMenuItems()` internally needs `VaadinService`. It gets it from the `VaadinRequest` (if given) or calling `VaadinService.getCurrent()`.
Since we do not have a `VaadinRequest`, we need to manually set the thread locale

```java
Map<Class<?>, CurrentInstance> instances = CurrentInstance.getInstances();
CurrentInstance.clearAll();
VaadinService.setCurrent(vaadinService);
try {
    setRoutes(MenuRegistry.collectClientMenuItems(false, config, null));
} finally {
    CurrentInstance.restoreInstances(instances);
}

```